### ISSUE TITLE
Add support for temporary rules

### ISSUE THREAD

**Issue by mmsatari** [salarcode/SmartProxy#116]
It would be nice if user could add temporary rules that would get deleted after browser restart. This is handy for some websites that users don't visit frequently and they want to use proxy on them.

**Comment by salarcode** [comment]
I don't much use scenario for these kind of rules. Do you have any sample where this kind of stuff is implemented?

**Comment by mmsatari** [comment]
FoxyProxy used to support this kind of rules. These rules existed until the user closes the Firefox process. I don't how it was implemented, but one possible solution would be to have two seperate DBs for each type of rule (permanent, temporary) and delete the latter when the Firefox process is being restarted.

**Comment by smsihe** [comment]
Yes , Tampermonkey just works this way  .

**Comment by shinji-ikari** [comment]
yes, i love switchyomega with this temporary proxy funtion, sometimes, it just no need to keep it in proxy rules.
i do use this funtion for some kind website that when i search something and it maybe don't have second time to visit.
also there are some website i use proxy and direct at the same time, by this when i done with direct then use proxy , and next time i restart the firefox, it will be direct in default. don't need to change the rules.
hope there are enough people want this funtion.

**and thanks to u guys, smartproxy is kind of good and easy to use.  since switchomega work on firefox is more and more inconvenient.

**Comment by tglsf** [comment]
> I don't see much use case scenario for these kind of rules. Do you have any samples where this kind of stuff is implemented?

This feature is very useful to me. Sometimes you want to determine if a site is truly inaccessible and need to toggle the proxy status. Once you have determined that it is an inaccessible site, you do not need to remove the rule by visiting the settings page if you are using a temporary rule. If you try to access those inaccessible sites from time to time, switch the proxy state and do not clean up your rule list, then the rule list will become very long.
Also, sometimes I want to log in to certain sites temporarily as anonymous, so this feature can be useful.

**Comment by leic4u** [comment]
It is necessary when I have multiple proxies and want to temporarily change it.
